app/src/main/java/de/danoeh/antennapod/fragment/EpisodesListFragment.java

```diff
@@ -214,6 +217,14 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
             ((SimpleItemAnimator) animator).setSupportsChangeAnimations(false);
         }
 
+        SwipeRefreshLayout swipeRefreshLayout = root.findViewById(R.id.swipeRefresh);
+        swipeRefreshLayout.setOnRefreshListener(() -> {
+            AutoUpdateManager.runImmediate(requireContext());
+            new Handler().postDelayed(() ->
+                            getActivity().runOnUiThread(() -> swipeRefreshLayout.setRefreshing(false))
```

Comment:
I think you don't need `getActivity().runOnUiThread`. Handlers that are created on the UI thread run their `runnable` on the UI thread, too.